File: perl/basics/fizzbuzz.pl
#!/usr/bin/env perl
# Print FizzBuzz for 1 through 100, choosing the label from the remainders.

use strict;
use warnings;

sub label {
    my ($n) = @_;
    return 'FizzBuzz' if $n % 15 == 0;
    return 'Fizz'     if $n % 3 == 0;
    return 'Buzz'     if $n % 5 == 0;
    return $n;
}

print label($_), "\n" for 1 .. 100;

File: clojure/basics/fizzbuzz.clj
;; Print FizzBuzz for 1 through 100, choosing the label with cond.
;; Run: clojure -M fizzbuzz.clj

(defn fizzbuzz-label [n]
  (cond
    (zero? (mod n 15)) "FizzBuzz"
    (zero? (mod n 3)) "Fizz"
    (zero? (mod n 5)) "Buzz"
    :else (str n)))

(doseq [n (range 1 101)]
  (println (fizzbuzz-label n)))
